it/controllers/benefits/income/IncomeTaxBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,546 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.benefits.income
+
+import forms.AmountForm
+import controllers.employment.routes.CheckYourBenefitsController
+import models.User
+import models.benefits.{BenefitsViewModel, IncomeTaxAndCostsModel}
+import models.mongo.{EmploymentCYAModel, EmploymentDetails, EmploymentUserData}
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.http.HeaderNames
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import play.api.libs.ws.WSResponse
+import utils.{EmploymentDatabaseHelper, IntegrationTest, ViewHelpers}
+import controllers.benefits.income.routes.IncomeTaxBenefitsAmountController
+
+class IncomeTaxBenefitsAmountControllerISpec extends IntegrationTest with ViewHelpers with EmploymentDatabaseHelper {
+
+  val taxYearEOY: Int = taxYear - 1
+  val employmentId: String = "001"
+  val amount: BigDecimal = 100
+  val amountFieldName = "amount"
+  val expectedErrorHref = "#amount"
+
+  def pageUrl(taxYear: Int): String =
+    s"$appUrl/$taxYear/benefits/employer-income-tax-amount?employmentId=$employmentId"
+
+  val formLink: String =
+    s"/update-and-submit-income-tax-return/employment-income/$taxYearEOY/benefits/employer-income-tax-amount?employmentId=$employmentId"
+
+  private val userRequest = User(mtditid, None, nino, sessionId, affinityGroup)(fakeRequest)
+
+  private def employmentUserData(hasPriorBenefits: Boolean, employmentCyaModel: EmploymentCYAModel): EmploymentUserData =
+    EmploymentUserData(sessionId, mtditid, nino, taxYearEOY, employmentId, isPriorSubmission = true, hasPriorBenefits = hasPriorBenefits, employmentCyaModel)
+
+
+  def cyaModel(benefits: Option[BenefitsViewModel] = None): EmploymentCYAModel =
+    EmploymentCYAModel(EmploymentDetails("employerName", currentDataIsHmrcHeld = true), benefits)
+
+  def benefits(incomeTaxModel: IncomeTaxAndCostsModel): BenefitsViewModel =
+    BenefitsViewModel(
+      carVanFuelModel = Some(fullCarVanFuelModel), accommodationRelocationModel = Some(fullAccommodationRelocationModel),
+      travelEntertainmentModel = Some(fullTravelOrEntertainmentModel), utilitiesAndServicesModel = Some(fullUtilitiesAndServicesModel),
+      medicalChildcareEducationModel = Some(fullMedicalChildcareEducationModel),
+      isUsingCustomerData = true, isBenefitsReceived = true, incomeTaxAndCostsModel = Some(incomeTaxModel))
+
+  object Selectors {
+    val captionSelector = "#main-content > div > div > form > div > label > header > p"
+    def paragraphSelector(index: Int): String = s"#main-content > div > div > form > div > label > p:nth-child($index)"//change using page inspect
```

Comment:
remove the comment on the end of this line please

Reply:
comment has been removed